typing/predef.ml

```diff
@@ -68,27 +139,37 @@ and path_string = Pident ident_string
 and path_extension_constructor = Pident ident_extension_constructor
 and path_floatarray = Pident ident_floatarray
 
-let type_int = newgenty (Tconstr(path_int, [], ref Mnil))
-and type_char = newgenty (Tconstr(path_char, [], ref Mnil))
-and type_bytes = newgenty (Tconstr(path_bytes, [], ref Mnil))
-and type_float = newgenty (Tconstr(path_float, [], ref Mnil))
-and type_bool = newgenty (Tconstr(path_bool, [], ref Mnil))
-and type_unit = newgenty (Tconstr(path_unit, [], ref Mnil))
-and type_exn = newgenty (Tconstr(path_exn, [], ref Mnil))
-and type_eff t = newgenty (Tconstr(path_eff, [t], ref Mnil))
-and type_continuation t1 t2 =
-  newgenty (Tconstr(path_continuation, [t1; t2], ref Mnil))
-and type_array t = newgenty (Tconstr(path_array, [t], ref Mnil))
-and type_list t = newgenty (Tconstr(path_list, [t], ref Mnil))
-and type_option t = newgenty (Tconstr(path_option, [t], ref Mnil))
-and type_nativeint = newgenty (Tconstr(path_nativeint, [], ref Mnil))
-and type_int32 = newgenty (Tconstr(path_int32, [], ref Mnil))
-and type_int64 = newgenty (Tconstr(path_int64, [], ref Mnil))
-and type_lazy_t t = newgenty (Tconstr(path_lazy_t, [t], ref Mnil))
-and type_string = newgenty (Tconstr(path_string, [], ref Mnil))
-and type_extension_constructor =
-      newgenty (Tconstr(path_extension_constructor, [], ref Mnil))
-and type_floatarray = newgenty (Tconstr(path_floatarray, [], ref Mnil))
+let path_of_type_constr typ =
+  Pident (ident_of_type_constr typ)
```

Comment:
Since there is only one usage of this helper function (in `find_type_constr`) I suggest moving it closer to its call site.

Reply:
My idea is that we keep the `ident_*` names for compatibility, and then the generic mechanism `ident_of_type_constr` which logically comes right after. Similarly, we keep `path_*` names, and then the generic `path_of_type_constr` function. So the place makes sense to me. But then that would suggest making those functions public -- exposing them in the .mli, which I haven't done yet, I'm not sure why. I guess I need to decide whether I consider this a useful function to expose, and then actually expose it (same with `ident_of_type_constr`), or we don't care about this function, and it could be inlined away as you point out. Do you have an opinion?

Reply:
I don't have a strong opinion, and agree with the reasoning behind the current placement of the `find_type_constr`.